src/services/expenses.services.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const { Expense } = require('../repository/db');
+const { Op } = require('sequelize');
+
+const findAll = async(filterProperties) => {
+  const { userId, categories, from, to } = filterProperties;
+
+  const whereClause = {};
+
+  whereClause.userId = userId;
+
+  if (categories) {
+    whereClause.category = Array.isArray(categories) && categories.length > 0
+      ? { [Op.in]: categories }
+      : categories;
+  }
+
+  if (from || to) {
+    whereClause.spentAt = {};
+
+    if (from) {
+      whereClause.spentAt[Op.gte] = new Date(from);
+    }
+
+    if (to) {
+      whereClause.spentAt[Op.lte] = new Date(to);
+    }
+  }
+
+  return Expense.findAll({ where: whereClause });
+};
+
+const getById = async(id) => {
+  return Expense.findByPk(+id);
+};
+
+const create = async(expense) => {
+  try {
+    const lastExpense = await Expense.findOne({
+      attributes: ['id'],
+      order: [['id', 'DESC']],
+    });
+
+    const getMaxId = lastExpense ? lastExpense.id : 0;
+
+    const newExpense = {
+      id: getMaxId + 1,
+      ...expense,
+    };
+
+    return Expense.create(newExpense);
+  } catch (error) {
+    throw error;
+  }
```

Comment:
- If you `catch` an error in the controller you don't need to wrap it here
- You can add autoincrement option for the `id` field, so you don't need to do it manually